tabular/src/autogluon/tabular/models/lgb/lgb_utils.py

```diff
@@ -94,3 +102,69 @@ def construct_dataset(x: DataFrame, y: Series, location=None, reference=None, pa
         # dataset_binary = lgb.Dataset(location + '.bin', reference=reference, free_raw_data=False)# .construct()
 
     return dataset
+
+
+def train_lgb_model(early_stopping_callback_kwargs=None, **train_params):
+    import lightgbm as lgb
+
+    if train_params['params']['objective'] == 'quantile':
+        quantile_levels = train_params['params'].pop('quantile_levels')
+        booster = QuantileBooster(quantile_levels=quantile_levels, early_stopping_callback_kwargs=early_stopping_callback_kwargs)
+        return booster.fit(**train_params)
+    else:
+        return lgb.train(**train_params)
+
+
+class QuantileBooster:
+    """Wrapper that trains a separate LGBM Booster for each quantile level."""
+    def __init__(self, quantile_levels: List[float], early_stopping_callback_kwargs: Optional[dict] = None):
+        if quantile_levels is None:
+            raise AssertionError
+        if not all(0 < q < 1 for q in quantile_levels):
+            raise AssertionError(f'quantile_levels must fulfill 0 < q < 1, provided quantile_levels: {quantile_levels}')
+
+        self.quantile_levels = quantile_levels
+
+        self.early_stopping_callback_kwargs = None
+        self.time_limit_global = None
+
+        if early_stopping_callback_kwargs is not None:
+            self.early_stopping_callback_kwargs = early_stopping_callback_kwargs
+            self.time_limit_global = early_stopping_callback_kwargs.pop('time_limit')
+        self.model_dict = {}
+
+    def fit(self, **train_params_base):
+        import lightgbm as lgb
+        from .callbacks import early_stopping_custom
+
+        start_time_global = time.time()
+
+        for q in self.quantile_levels:
+            train_params = copy.deepcopy(train_params_base)
+            train_params['params']['alpha'] = q
+            if self.early_stopping_callback_kwargs is not None:
+                es_kwargs = copy.deepcopy(self.early_stopping_callback_kwargs)
+                if self.time_limit_global is not None:
+                    es_kwargs['start_time'] = time.time()
+                    es_kwargs['time_limit'] = self.time_limit_global / len(self.quantile_levels)
+                train_params['callbacks'].append(early_stopping_custom(**es_kwargs))
```

Comment:
train_params already has the callback, why are we adding a second one and not removing the first?